### ISSUE TITLE
Security Testing on a remote checks the wrong remote if an earlier remote is destroyed

### ISSUE THREAD

**Issue by anathomical** [mtgred/netrunner#742]
If you have Security Testing installed at the beginning of your turn, and select Server 1 as its target, then you run Server 0 which has no ice and trash whatever is in it, Server 1 falls into the position of Server 0.  If you run the new Server 0, Security testing will not trigger.  If you then run Server 1 (assuming that there was a Server 2 at the start of the turn to fall into this position), then Security Testing will trigger rather than allowing you access.

Basically, it looks like Security Testing ties itself to a server index when you select at the beginning of turn, but that index is not updated with the destruction of servers which may change the contents of the server at that index.

**Comment by JoelCFC25** [comment]
Server destruction is definitely a problem. This same issue plagues Franchise City (or any asset/upgrade that might disappear or be trashed mid-run)--if you have one rezzed when an agenda is accessed in a higher-numbered remote, the Runner's steal prompt for the agenda is bugged because of collapsing servers and the steal cannot be completed.

**Comment by queueseven** [comment]
I'll try to fix this tomorrow if I find the time (and fixing means removing the server destruction code. Too...many...problems....)

**Comment by nealterrell** [comment]
Yeah, this was noted when ST was implemented, but I didn't know how to fix it at the time. Sounds like @queueseven is on it!
